builder/processor/src/main/java/io/helidon/builder/processor/BlueprintProcessor.java

```diff
@@ -253,331 +259,223 @@ private void generatePrototypeWithBuilder(TypeElement builderInterface,
-                if (!blueprintDef.typeArguments().isEmpty()) {
-                    pw.print(" ");
-                    pw.print(blueprintDef.typeArguments());
+        // prototype interface (with inner class Builder)
+        ClassModel.Builder classModel = ClassModel.builder()
+                .type(prototype)
+                .classType(ClassType.INTERFACE)
+                .licenseHeader(CopyrightHandler.copyright(GENERATOR,
+                                                          typeInfo.typeName(),
+                                                          prototype));
+
+        String javadocString = blueprintDef.javadoc();
+        List<TypeArgument> typeArguments = new ArrayList<>();
+        if (javadocString == null) {
+            classModel.description("Interface generated from definition. Please add javadoc to the definition interface.");
+            typeGenericArguments.forEach(arg -> typeArguments.add(TypeArgument.builder()
+                                      .token(arg.className())
+                                      .build()));
+        } else {
+            Javadoc javadoc = Javadoc.parse(blueprintDef.javadoc());
+            classModel.javadoc(javadoc);
+            typeGenericArguments.forEach(arg -> {
+                TypeArgument.Builder tokenBuilder = TypeArgument.builder().token(arg.className());
+                if (javadoc.genericsTokens().containsKey(arg.className())) {
+                    tokenBuilder.description(javadoc.genericsTokens().get(arg.className()));
                 }
-                pw.print(" Builder");
-                pw.print(blueprintDef.typeArguments());
-                pw.print(" builder(");
-                pw.print(ifaceName);
-                pw.print(blueprintDef.typeArguments());
-                pw.println(" instance) {");
-                pw.print(SOURCE_SPACING);
-                pw.print(SOURCE_SPACING);
-                pw.print("return ");
-                pw.print(ifaceName);
-                pw.print(".");
-                pw.print(blueprintDef.typeArguments());
-                pw.println("builder().from(instance);");
-                pw.print(SOURCE_SPACING);
-                pw.println("}");
-                pw.println();
-            }
+                typeArguments.add(tokenBuilder.build());
+            });
+        }
+        typeArguments.forEach(classModel::addGenericArgument);
 
-            /*
-             static X create(Config config)
-             */
-            if (blueprintDef.createFromConfigPublic() && configuredData.configured()) {
-                pw.print(SOURCE_SPACING);
-                pw.println("/**");
-                pw.print(SOURCE_SPACING);
-                pw.println(" * Create a new instance from configuration.");
-                pw.print(SOURCE_SPACING);
-                pw.println(" *");
-                pw.print(SOURCE_SPACING);
-                pw.println(" * @param config used to configure the new instance");
```

Comment:
Are you verifying the javadoc for this (params, return, etc.) are all same as before?

Reply:
Hmm, I am not sure what do you mean exactly. Could you please elaborate? :-)

Reply:
did you compare the generated code to the original to ensure it all looks the same?